contracts/libraries/VaultLifecycleSTETH.sol

```diff
@@ -0,0 +1,747 @@
+            // Unrap to stETH
+            wsteth.unwrap(wsteth.balanceOf(address(this)));
+
+            // CRV SWAP HERE from steth -> eth
+        }
+    }
+
+    /**
+     * @notice Wraps the necessary amount of the base token to the yield-bearing yearn token
+     * @param collateralToken is the address of the collateral token
+     */
+    function wrapToYieldToken(address collateralToken) external {
+        ISTETH steth = ISTETH(IWSTETH(collateralToken).stETH());
+        if (address(this).balance > 0) {
+            // Send eth to Lido, recieve steth
+            steth.submit{value: address(this).balance}(address(this));
+            // Wrap to wstETH
+            IWSTETH(collateralToken).wrap(steth.balanceOf(address(this)));
+        }
+    }
+
+    function getVaultFees(
+        Vault.VaultState storage vaultState,
+        uint256 currentLockedBalance,
+        uint256 performanceFeePercent,
+        uint256 managementFeePercent
+    )
+        external
+        view
+        returns (
+            uint256 performanceFee,
+            uint256 managementFee,
+            uint256 vaultFee
+        )
+    {
+        uint256 prevLockedAmount = vaultState.lastLockedAmount;
+        uint256 totalPending = vaultState.totalPending;
+
+        // Take performance fee and management fee ONLY if difference between
+        // last week and this week's vault deposits, taking into account pending
+        // deposits and withdrawals, is positive. If it is negative, last week's
+        // option expired ITM past breakeven, and the vault took a loss so we
+        // do not collect performance fee for last week
+        if (currentLockedBalance.sub(totalPending) > prevLockedAmount) {
+            performanceFee = currentLockedBalance
+                .sub(totalPending)
+                .sub(prevLockedAmount)
+                .mul(performanceFeePercent)
+                .div(100 * 10**6);
+            managementFee = currentLockedBalance.mul(managementFeePercent).div(
+                100 * 10**6
+            );
+
+            vaultFee = performanceFee.add(managementFee);
+        }
+    }
+
+    /**
+     * @notice Helper function to make either an ETH transfer or ERC20 transfer
+     * @param weth is the weth address
+     * @param asset is the vault asset address
+     * @param recipient is the receiving address
+     * @param amount is the transfer amount
+     */
+    function transferAsset(
+        address weth,
+        address asset,
+        address payable recipient,
+        uint256 amount
+    ) public {
+        if (asset == weth) {
+            IWETH(weth).withdraw(amount);
+            (bool success, ) = recipient.call{value: amount}("");
+            require(success, "!success");
+            return;
+        }
+        IERC20(asset).safeTransfer(recipient, amount);
+    }
+
+    /**
+     * @notice Gets the next options expiry timestamp
+     */
+    function getNextFriday(uint256 currentExpiry)
```

Comment:
This function needs to be copy paste the updated version from VaultLifecycle.sol

Reply:
yeah, want to do this all in separate pr for leftover chainsafe changes to be addressed :)